### PR TITLE
Further Code Clean Up

### PR DESCRIPTION
**Pull request by MartinX3**
Added ToDo's

Fixed moons-String.

Optimized Code further (C#7 conventions)                              

Changed some code to LINQ

### CODE REVIEW

File: Source/Database/PQSPreset.cs [hunk 1]
```diff
@@ -4,19 +4,23 @@
  * Licensed under the Terms of the MIT License
  */
 
-using System;
-using ConfigNodeParser;
-using Kopernicus.Configuration;
-
 namespace Stellarator.Database
 {
+    using ConfigNodeParser;
```
Comment (StollD) [inline Source/Database/PQSPreset.cs]: Please leave the usings outside of the namespace declaration.

File: Source/Database/PQSPreset.cs [hunk 2]
```diff
@@ -4,19 +4,23 @@
...
+    using Kopernicus.Configuration;
+
     /// <summary>
-    /// Prefab for a PQS
+    ///     Prefab for a PQS
```
Comment (StollD) [inline Source/Database/PQSPreset.cs]: I am not sure if I like these leading whitespaces.

File: Source/Generator.cs [hunk 3]
```diff
@@ -4,179 +4,189 @@
...
         /// </summary>
-        public static Random Random { get; set; }
+        public static Random Random { get; private set; }
 
         /// <summary>
-        /// The seed the system is using
+        ///     The seed the system is using
         /// </summary>
-        public static Int32 Seed { get; set; }
+        private static int Seed { get; set; }
```
Comment (StollD) [inline Source/Generator.cs]: Please use CLR type names instead of type aliases

File: Source/Generator.cs [hunk 4]
```diff
@@ -4,179 +4,189 @@
...
+        public static void Generate(string seed, string folder, bool systematicNames)
         {
             // Log
             Console.WriteLine("Generating the solar system...");
             Console.WriteLine();
 
             // Create a new Solar System using libaccrete
-            SolarSystem system = new SolarSystem(false, true, s => { });
-            SolarSystem.Generate(ref system, seed.GetHashCode(), Int32.MaxValue);
+            var system = new SolarSystem(false, true, s => { });
```
Comment (StollD) [inline Source/Generator.cs]: Please don't use var.

File: Source/Generator.cs [hunk 5]
```diff
@@ -209,70 +222,81 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
...
             {
                 if (planet.gas_giant)
+                {
                     template.AddValue("name", "Jool");
+                }
                 else
                 {
                     template.AddValue("name", Utility.GetTemplate(planet.surface_pressure > 0.00001, false));
-                    template.AddValue("removeAllPQSMods", "True");
+                    template.AddValue("removeAllPQSmods", "True");
```
Comment (StollD) [inline Source/Generator.cs]: You know that you changed a value that is later read by Kopernicus? Kopernicus is case sensitive

File: Source/Generator.cs [hunk 6]
```diff
@@ -357,18 +387,19 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
             // PQS
             if (!planet.gas_giant)
             {
-                Color average = default(Color);
-                GeneratePQS(ref node, name, folder, planet, ref average);
+                Color average;
+                GeneratePqs(ref node, name, folder, planet, out average);
```
Comment (StollD) [inline Source/Generator.cs]: Please keep it PQS

File: Source/Generator.cs [hunk 7]
```diff
@@ -384,173 +415,204 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
...
             // Create the node
-            ConfigNode pqs = new ConfigNode("PQS");
+            var pqs = new ConfigNode("PQS");
 
-            // TODO: Material Settings?
+            // TODO: material Settings?
 
             // Create a node for the mods
-            ConfigNode mods = new ConfigNode("Mods");
+            var mods = new ConfigNode("mods");
```
Comment (StollD) [inline Source/Generator.cs]: Same as above, you changed a value that is later read by Kopernicus

File: Source/Generator.cs [hunk 8]
```diff
@@ -384,173 +415,204 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
...
-                    height.SetPixel(i, j, new Color(h, h, h));
-                }
-            }
-            diffuse.UnlockBitmap();
-            height.UnlockBitmap();
+                        for (var j = 0; j < (width / 2); j++)
+                        {
+                            // Create a VertexBuildData
+                            var builddata = new VertexBuildData
+                                            {
```
Comment (StollD) [inline Source/Generator.cs]: This indent looks weird

File: Source/Generator.cs [hunk 9]
```diff
@@ -384,173 +415,204 @@ public static ConfigNode GenerateBody(Planet planet, String folder, String refer
...
-            // Save the textures
-            Directory.CreateDirectory(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/");
-            UnsafeBitmap normals = Utility.BumpToNormalMap(height, 9); // TODO: Implement something to make strength dynamic
-            diffuse.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Texture.png", ImageFormat.Png);
-            height.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Height.png", ImageFormat.Png); // In case you need it :)
-            normals.Bitmap.Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Normals.png", ImageFormat.Png);
+                    // Save the textures
+                    Directory.CreateDirectory(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/");
+                    using (var normals = Utility.BumpToNormalMap(height, 9)
+                    ) // TODO: Implement something to make strength dynamic
```
Comment (StollD) [inline Source/Generator.cs]: No newline maybe?

**Review by StollD — CHANGES_REQUESTED** [review]

File: Source/Database/PQSPreset.cs (hunk 2 above)
Comment (MartinX3) [reply Source/Database/PQSPreset.cs]: Should I revert this?
I thought with them, the text between the next nodes is easier to read. :)

File: Source/Generator.cs (hunk 9 above)
Comment (MartinX3) [reply Source/Generator.cs]: The newline intends a better reading
Should I revert it?

```
diffuse.Bitmap
       .Save(Directory.GetCurrentDirectory() + "/systems/" + folder + "/PluginData/" + name + "_Texture.png",
             ImageFormat.Png);
```

File: Source/Generator.cs (hunk 8 above)
Comment (MartinX3) [reply Source/Generator.cs]: ReSharper told it to me, for a better readability
Should i revert it?

**Comment by MartinX3** [comment]
I made several commits to integrate your advises
I look how I can merge them with this pull request.

Edit:
Oh, they are automatically integrated. Nice.
@StollD may you look at this?

File: Source/Generator.cs (hunk 9 above)
Comment (StollD) [reply Source/Generator.cs]: In the code I marked, there is a new line for a `)`, I don't think that is neccessary. The comment can stay on the new line though.

File: Source/Program.cs [hunk 10]
```diff
@@ -37,9 +37,9 @@ public static int Main(string[] args)
 
             //TODO: MartinX3 Change How we get the args. Only show question, if arg is not given.
             // Ask for Input
-            var seed = Prompt("Please enter the seed you want to use: ", "--seed");
-            var folder = Prompt("Please choose a folder name for your system: ", "--name");
-            var systematic = Prompt("Use systematic planet names? (y/n) ", "--systematic", true);
+            string seed = Prompt("Please enter the seed you want to use: ", "--seed");
```
Comment (StollD) [inline Source/Program.cs]: Please use CLR type names here too (and everywhere else)

**Comment by MartinX3** [comment]
Hey @StollD , is that okay now? :)

File: Source/Generator.cs (hunk 9 above)
Comment (MartinX3) [reply Source/Generator.cs]: Ahh, It is changed. :)

**Comment by MartinX3** [comment]
I also added an Info about the code conventions, you want in this project. :)

**Comment by StollD** [comment]
Looks ok to me. Merging.

Thanks for your contribution!

**Review by StollD — APPROVED** [review]

**Comment by MartinX3** [comment]
Perfect :)
And I'm glad, that you are happy about my contribution :D

**Pull request merged**